### ISSUE TITLE
KIE Sandbox not evaluating "Priority" Hit Policy correctly

### ISSUE THREAD

**Issue by jeremyfordip** [apache/incubator-kie-issues#985]
Issue: I am unable to build and verify a DMN that uses a "Priority" Hit Policy using the KIE Sandbox version 0.32.0.

The intention was to implement this logic:
<img width="458" alt="Screenshot 2024-03-04 at 3 58 53 pm" src="https://github.com/apache/incubator-kie-issues/assets/3805536/0144bfa8-c7cb-401e-8607-5cc0b934e163">

The implementation I have on the Sandbox is:
![Screenshot 2024-03-04 at 3 59 55 pm](https://github.com/apache/incubator-kie-issues/assets/3805536/9f0cd039-e9b4-40ac-b3bf-d53a97473208)

The results in the above table are incorrect. The first and last result should be "Declined".

When I take the generated DMN XML and test it using the Java evaluator it evaluates as expected.

```
<dependency>
      <groupId>org.kie.kogito</groupId>
      <artifactId>kogito-dmn</artifactId>
      <version>8.45.0-SNAPSHOT</version>
</dependency>
```     

The generated DMN is:
```
<?xml version="1.0" encoding="UTF-8"?>
<dmn:definitions xmlns:dmn="http://www.omg.org/spec/DMN/20180521/MODEL/" xmlns="https://kiegroup.org/dmn/_9FC3E933-E481-49A1-9F78-3ABEB8740D09" xmlns:feel="http://www.omg.org/spec/DMN/20180521/FEEL/" xmlns:kie="http://www.drools.org/kie/dmn/1.2" xmlns:dmndi="http://www.omg.org/spec/DMN/20180521/DMNDI/" xmlns:di="http://www.omg.org/spec/DMN/20180521/DI/" xmlns:dc="http://www.omg.org/spec/DMN/20180521/DC/" id="_37436097-C556-4E05-9BCB-306F47F2F4EF" name="Untitled" typeLanguage="http://www.omg.org/spec/DMN/20180521/FEEL/" namespace="https://kiegroup.org/dmn/_9FC3E933-E481-49A1-9F78-3ABEB8740D09">
  <dmn:extensionElements/>
  <dmn:inputData id="_581684A5-2D9E-411E-B731-6C398F948007" name="CreditRiskCategory">
    <dmn:extensionElements/>
    <dmn:variable id="_0D993DFE-D902-435A-B7C1-4685C31EABD0" name="CreditRiskCategory" typeRef="string"/>
  </dmn:inputData>
  <dmn:inputData id="_0A11B14E-CC51-4EA0-B4C5-B930D6818A0C" name="AffordabilityCategory">
    <dmn:extensionElements/>
    <dmn:variable id="_143126F1-A053-48EB-A9FC-FF65F5549922" name="AffordabilityCategory" typeRef="string"/>
  </dmn:inputData>
  <dmn:decision id="_31C9E2FF-8A75-474A-8342-5B99BDF124C1" name="LoanApproval">
    <dmn:extensionElements/>
    <dmn:allowedAnswers>"Declined", "Approved"</dmn:allowedAnswers>
    <dmn:variable id="_19FA2419-8E4E-4BBC-B580-C307273F9880" name="LoanApproval" typeRef="string"/>
    <dmn:informationRequirement id="_BA0F4D08-913C-4F26-966E-C5372D8BCAC5">
      <dmn:requiredInput href="#_581684A5-2D9E-411E-B731-6C398F948007"/>
    </dmn:informationRequirement>
    <dmn:informationRequirement id="_D380B11F-A98A-4712-B2E4-027D366BB289">
      <dmn:requiredInput href="#_0A11B14E-CC51-4EA0-B4C5-B930D6818A0C"/>
    </dmn:informationRequirement>
    <dmn:decisionTable id="_BD741981-CDBE-467B-961A-0992AAE097F1" hitPolicy="PRIORITY" preferredOrientation="Rule-as-Row">
      <dmn:input id="_424B8D7A-22DE-4DB1-9995-251EC43F3B5B">
        <dmn:inputExpression id="_7E14207A-A5A6-4AD0-90D2-683D591AFDED" typeRef="string">
          <dmn:text>AffordabilityCategory</dmn:text>
        </dmn:inputExpression>
      </dmn:input>
      <dmn:input id="_9D1E5D4A-78B3-4A64-A9A5-81527DE6CF33">
        <dmn:inputExpression id="_C72911ED-FA0E-43F0-B08B-3BB75455E1F9" typeRef="string">
          <dmn:text>CreditRiskCategory</dmn:text>
        </dmn:inputExpression>
      </dmn:input>
      <dmn:output id="_6D0EA2DE-7115-4F36-8C5B-F5AD27C6F0C2">
        <dmn:outputValues id="_F9AF738C-5E2A-439D-8725-678DE1C1ED54">
          <dmn:text>"Declined", "Approved"</dmn:text>
        </dmn:outputValues>
      </dmn:output>
      <dmn:annotation name="annotation-1"/>
      <dmn:rule id="_5A4392E6-1341-46D2-BCFC-84E163F88024">
        <dmn:inputEntry id="_9CDCB1FA-CE0A-45B0-B43C-8809F8F7BDB6">
          <dmn:text>-</dmn:text>
        </dmn:inputEntry>
        <dmn:inputEntry id="_F80DB69C-F17B-4BA0-A4CD-527431296708">
          <dmn:text>"High"</dmn:text>
        </dmn:inputEntry>
        <dmn:outputEntry id="_EA2C9C76-04B0-4974-B327-7CAD0E9BD97F">
          <dmn:text>"Declined"</dmn:text>
        </dmn:outputEntry>
        <dmn:annotationEntry>
          <dmn:text/>
        </dmn:annotationEntry>
      </dmn:rule>
      <dmn:rule id="_B30ED83D-9991-4E28-8167-90C1ED9233BE">
        <dmn:inputEntry id="_AA192B9D-CC36-4D66-8BAE-5AD493961300">
          <dmn:text>"Unaffordable"</dmn:text>
        </dmn:inputEntry>
        <dmn:inputEntry id="_11B38B87-4B69-42AC-80B5-FD2A227F163E">
          <dmn:text>-</dmn:text>
        </dmn:inputEntry>
        <dmn:outputEntry id="_0D55605C-1EEB-43DA-8E14-813065EE8D40">
          <dmn:text>"Declined"</dmn:text>
        </dmn:outputEntry>
        <dmn:annotationEntry>
          <dmn:text/>
        </dmn:annotationEntry>
      </dmn:rule>
      <dmn:rule id="_D8B4FB57-F7FD-41F9-9A0D-771765BD4B2B">
        <dmn:inputEntry id="_A6264040-1094-42B9-9E01-E5104AFE9BAC">
          <dmn:text>-</dmn:text>
        </dmn:inputEntry>
        <dmn:inputEntry id="_D659BD0C-DB9F-459D-AD0D-C157F6495B76">
          <dmn:text>-</dmn:text>
        </dmn:inputEntry>
        <dmn:outputEntry id="_0C6A14AE-8458-4ABD-A7E5-D243E6099FD6">
          <dmn:text>"Approved"</dmn:text>
        </dmn:outputEntry>
        <dmn:annotationEntry>
          <dmn:text/>
        </dmn:annotationEntry>
      </dmn:rule>
    </dmn:decisionTable>
  </dmn:decision>
  <dmndi:DMNDI>
    <dmndi:DMNDiagram id="_F16776B3-F579-481E-A027-11A770A9A50F" name="DRG">
      <di:extension>
        <kie:ComponentsWidthsExtension>
          <kie:ComponentWidths dmnElementRef="_BD741981-CDBE-467B-961A-0992AAE097F1">
            <kie:width>50</kie:width>
            <kie:width>237</kie:width>
            <kie:width>281</kie:width>
            <kie:width>295</kie:width>
            <kie:width>100</kie:width>
          </kie:ComponentWidths>
        </kie:ComponentsWidthsExtension>
      </di:extension>
      <dmndi:DMNShape id="dmnshape-drg-_581684A5-2D9E-411E-B731-6C398F948007" dmnElementRef="_581684A5-2D9E-411E-B731-6C398F948007" isCollapsed="false">
        <dmndi:DMNStyle>
          <dmndi:FillColor red="255" green="255" blue="255"/>
          <dmndi:StrokeColor red="0" green="0" blue="0"/>
          <dmndi:FontColor red="0" green="0" blue="0"/>
        </dmndi:DMNStyle>
        <dc:Bounds x="730" y="269" width="201" height="71"/>
        <dmndi:DMNLabel/>
      </dmndi:DMNShape>
      <dmndi:DMNShape id="dmnshape-drg-_0A11B14E-CC51-4EA0-B4C5-B930D6818A0C" dmnElementRef="_0A11B14E-CC51-4EA0-B4C5-B930D6818A0C" isCollapsed="false">
        <dmndi:DMNStyle>
          <dmndi:FillColor red="255" green="255" blue="255"/>
          <dmndi:StrokeColor red="0" green="0" blue="0"/>
          <dmndi:FontColor red="0" green="0" blue="0"/>
        </dmndi:DMNStyle>
        <dc:Bounds x="1123" y="278" width="237" height="62"/>
        <dmndi:DMNLabel/>
      </dmndi:DMNShape>
      <dmndi:DMNShape id="dmnshape-drg-_31C9E2FF-8A75-474A-8342-5B99BDF124C1" dmnElementRef="_31C9E2FF-8A75-474A-8342-5B99BDF124C1" isCollapsed="false">
        <dmndi:DMNStyle>
          <dmndi:FillColor red="255" green="255" blue="255"/>
          <dmndi:StrokeColor red="0" green="0" blue="0"/>
          <dmndi:FontColor red="0" green="0" blue="0"/>
        </dmndi:DMNStyle>
        <dc:Bounds x="914.5" y="455" width="229" height="83"/>
        <dmndi:DMNLabel/>
      </dmndi:DMNShape>
      <dmndi:DMNEdge id="dmnedge-drg-_BA0F4D08-913C-4F26-966E-C5372D8BCAC5-AUTO-TARGET" dmnElementRef="_BA0F4D08-913C-4F26-966E-C5372D8BCAC5">
        <di:waypoint x="830.5" y="304.5"/>
        <di:waypoint x="1029" y="455"/>
      </dmndi:DMNEdge>
      <dmndi:DMNEdge id="dmnedge-drg-_D380B11F-A98A-4712-B2E4-027D366BB289-AUTO-TARGET" dmnElementRef="_D380B11F-A98A-4712-B2E4-027D366BB289">
        <di:waypoint x="1241.5" y="309"/>
        <di:waypoint x="1029" y="455"/>
      </dmndi:DMNEdge>
    </dmndi:DMNDiagram>
  </dmndi:DMNDI>
</dmn:definitions>

```

**Comment by gitgabrio** [comment]
Hi @jeremyfordip 
I tested the DMN directly in the [engine](https://github.com/gitgabrio/drools/blob/80990367d6f29cb7135f7e8988f3335d8f702c91/kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java#L102) and through the jitexecutor.
Provided correct input data, the results are as expected.

So, it would be interesting to understand which data is actually sent to kie-sandbox.

@tiagobento FYI

**Comment by jomarko** [comment]
@jeremyfordip hi, I have one observation, that should not probably has an impact for the **Priority** hit policy, however it may be worth checking on your machine.

In the first screenshot your share, there are defined also column constraints, i.e. values that are acceptable as inputs for given columns. I speak about values for **Credit risk category** (High, Medium, Low) and **Affordability category** (Affordable, Marginal, Unaffordable)

While in the generated xml, it seems you do not set such constraint, because there is missing similar piece of xml:
```
      <dmn:input id="_9D1E5D4A-78B3-4A64-A9A5-81527DE6CF33">
        <dmn:inputExpression id="_C72911ED-FA0E-43F0-B08B-3BB75455E1F9" typeRef="string">
          <dmn:text>CreditRiskCategory</dmn:text>
        </dmn:inputExpression>
        <dmn:inputValues kie:constraintType="enumeration" id="_BFE965CA-6892-4DC9-9415-F22DC3E89376">
          <dmn:text>"High", "Medium", "Low"</dmn:text>
        </dmn:inputValues>
      </dmn:input>
```

Your model contains only:
```
      <dmn:input id="_9D1E5D4A-78B3-4A64-A9A5-81527DE6CF33">
        <dmn:inputExpression id="_C72911ED-FA0E-43F0-B08B-3BB75455E1F9" typeRef="string">
          <dmn:text>CreditRiskCategory</dmn:text>
        </dmn:inputExpression>
      </dmn:input>
```

Or in the UI editor, it look like this.
![Screenshot 2024-03-28 092007](https://github.com/apache/incubator-kie-issues/assets/8044780/94765704-3a54-427e-99de-52a14fe7bdf7)

**Comment by yesamer** [comment]
@jeremyfordip Can you please give us feedback considering the above observations? Thx.

**Comment by jeremyfordip** [comment]
@yesamer the feed back is that I'm a complete numpty and I'm sorry for wasting your time :(

I had copied the values from the Decision table to the Run Table, which included the quotes.

As soon as I removed them it was fine.  Adding constraints helped as it showed in the network response from the Extended Services that the value (with quotes) wasn't accepted.

Working as expected:
![Screenshot 2024-04-19 at 2 33 13 pm](https://github.com/apache/incubator-kie-issues/assets/3805536/c88c8a80-4de5-426b-96de-5358b920a66a)
